src/Middleware/HttpLogging/src/W3CLoggerProcessor.cs

```diff
@@ -20,97 +20,117 @@ public W3CLoggerProcessor(IOptionsMonitor<W3CLoggerOptions> options, IHostEnviro
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.ServerName))
         {
-            sb.Append(" s-computername");
+            await WriteMessageAsync(" s-computername", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.ServerIpAddress))
         {
-            sb.Append(" s-ip");
+            await WriteMessageAsync(" s-ip", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.ServerPort))
         {
-            sb.Append(" s-port");
+            await WriteMessageAsync(" s-port", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.Method))
         {
-            sb.Append(" cs-method");
+            await WriteMessageAsync(" cs-method", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.UriStem))
         {
-            sb.Append(" cs-uri-stem");
+            await WriteMessageAsync(" cs-uri-stem", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.UriQuery))
         {
-            sb.Append(" cs-uri-query");
+            await WriteMessageAsync(" cs-uri-query", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.ProtocolStatus))
         {
-            sb.Append(" sc-status");
+            await WriteMessageAsync(" sc-status", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.TimeTaken))
         {
-            sb.Append(" time-taken");
+            await WriteMessageAsync(" time-taken", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.ProtocolVersion))
         {
-            sb.Append(" cs-version");
+            await WriteMessageAsync(" cs-version", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.Host))
         {
-            sb.Append(" cs-host");
+            await WriteMessageAsync(" cs-host", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.UserAgent))
         {
-            sb.Append(" cs(User-Agent)");
+            await WriteMessageAsync(" cs(User-Agent)", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.Cookie))
         {
-            sb.Append(" cs(Cookie)");
+            await WriteMessageAsync(" cs(Cookie)", streamWriter, cancellationToken);
         }
         if (_loggingFields.HasFlag(W3CLoggingFields.Referer))
         {
-            sb.Append(" cs(Referer)");
+            await WriteMessageAsync(" cs(Referer)", streamWriter, cancellationToken);
         }
 
-        return sb.ToString();
+        // Send of line (only for test)
+        await WriteMessageAsync(string.Empty, streamWriter, cancellationToken, true);
```

Comment:
This shouldn't be in the non-test version of `W3CLoggerProcessor`, if the empty line is only for tests, it should only happen in tests